src/main/webapp/SearchResults.js

```diff
@@ -1,18 +1,32 @@
 /**
  * @param {query} input
  */
-function searching(input) {
+async function searching(input) {
   input = input.toLowerCase();
+  var countryId = null;
+  var countryName = null;
+
+  const countriesResponse = await fetch('/api/countries');
+  const countries = await countriesResponse.json();
+  
+  for (let country in countries) {
+    if(countries[country]['name'].toLowerCase() === input){
+        displayCountryResults(countries[country]);
+        countryId = countries[country]['id'];
+        countryName = countries[country]['name'];
+        break;
+    }
+  }
+
+  const citiesResponse = await fetch('/api/cities?name=' + input);
+  const cities = await citiesResponse.json();
 
-  if (input === 'south korea') {
-    southKoreaResults();
-  } else if (input === 'brazil') {
-    brazilResults();
-    console.log('true - brazil');
-  } else if (input === 'malaysia') {
-    malaysiaResults();
-    console.log('true - malaysia');
+  for (let city in cities) {
+    if(cities[city]['countryId'] !== countryId){
+        delete cities[city]; 
+    }
   }
```

Comment:
In this case, all cities that contain the value of `input` are returned but then they're filtered out based on the country ID. While I think it would be valuable to show all the cities located in a country whose name matches/contains the query, shouldn't we also include cities whose names match/include the query? E.g. the query `New` might return both the country `New Zealand` as well as the city `New York` which is not located in a country with the word `New` in the name